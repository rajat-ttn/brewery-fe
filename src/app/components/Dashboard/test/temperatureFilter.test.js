import React from 'react';
import { shallow } from 'enzyme';
import TemperatureFilter from '../temperatureFilter';

describe('Temperature Filter filters beer based on temperature correctly', () => {
    let allProps,
        temperatureFilter;
    const temperatureType = 'CELSIUS';
    const event = {
        preventDefault() {},
        target: { value: 'FAHRENHEIT' }
    };
    beforeEach(() => {
        allProps = {
            temperatureType: "FAHRENHEIT",
            changeTemperatureType: () => {}
        };
        temperatureFilter = shallow(<TemperatureFilter {...allProps} />);
    });

    it('temperature select should work properly', () => {
        expect(temperatureFilter).toMatchSnapshot()
    });

    it('trigger Event', () => {
        temperatureFilter.find('select').simulate('change', event);
        expect(temperatureFilter.find('select').props().value).toBe("FAHRENHEIT");
    });
});
