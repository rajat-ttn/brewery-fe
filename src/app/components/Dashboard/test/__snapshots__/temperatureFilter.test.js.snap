// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature Filter filters beer based on temperature correctly temperature select should work properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TemperatureFilter
    changeTemperatureType={[Function]}
    temperatureType="FAHRENHEIT"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Dropdown
        baseClassName="Dropdown"
        onChange={[Function]}
        options={
                Array [
                        Object {
                          "label": "Celsius",
                          "value": "CELSIUS",
                        },
                        Object {
                          "label": "Fahrenheit",
                          "value": "FAHRENHEIT",
                        },
                      ]
        }
        value="Fahrenheit"
/>,
      "className": "dropdownSection",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "baseClassName": "Dropdown",
        "onChange": [Function],
        "options": Array [
          Object {
            "label": "Celsius",
            "value": "CELSIUS",
          },
          Object {
            "label": "Fahrenheit",
            "value": "FAHRENHEIT",
          },
        ],
        "value": "Fahrenheit",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Dropdown
          baseClassName="Dropdown"
          onChange={[Function]}
          options={
                    Array [
                              Object {
                                "label": "Celsius",
                                "value": "CELSIUS",
                              },
                              Object {
                                "label": "Fahrenheit",
                                "value": "FAHRENHEIT",
                              },
                            ]
          }
          value="Fahrenheit"
/>,
        "className": "dropdownSection",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "baseClassName": "Dropdown",
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "Celsius",
              "value": "CELSIUS",
            },
            Object {
              "label": "Fahrenheit",
              "value": "FAHRENHEIT",
            },
          ],
          "value": "Fahrenheit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
